### PR TITLE
[#144035107] Upgrade CF to version 257

### PR DESCRIPTION
**Pull request by LeePorte**
## What

Story: [Upgrade CF to version >= 257](https://www.pivotaltracker.com/story/show/144035107)

Deploys upgraded Cloud Foundry (257). This also includes an upgrade to Diago (1.13.0) and Garden-Runc (1.5.0).
No upgrades to stemcels or buildpacks are required for this release.

This PR keeps the patched version of UAA in and also reverts the large file transfer SSH test workaround.

## How to review

1. Code review
2. Deploy from the branch
3. Rotate credentials - don't need to go out of `dev` environment

## Who can review
Not @paroxp or me

### CODE REVIEW

File: platform-tests/upstream/run_acceptance_tests.sh [hunk 1]
```diff
@@ -30,11 +30,6 @@ for i in $(seq $SLEEPTIME 1); do echo -ne "$i"'\r'; sleep 1; done; echo
...
 cd "${CF_GOPATH}/cf-acceptance-tests"
 
-# FIXME: Remove this once we've upgraded to CF 257+
-if [ "$(git rev-parse HEAD)" != "8fcde18d9b514fcf695f10049880aabe32910eb5" ]; then
-  echo "Unexpected revision of acceptance tests repo. Check FIXME in ${BASH_SOURCE[0]}"
-  exit 1
-fi
 # Checkout SHA of merge with updates for cf-cli 6.26
 # This is the only change from the existing commit.
 git checkout 407abdc
```
Comment (alext) [inline platform-tests/upstream/run_acceptance_tests.sh]: This isn't a full revert of ab15b9537. This `git checkout` line shouldn't remain here.

The best thing to do here would be a `git revert ab15b9537`. It shouldn't need any modification.

Comment (LeePorte) [inline platform-tests/upstream/run_acceptance_tests.sh]: Comment addressed

**Comment by alext** [comment]
I see changes to uaa permissions in the upstream templates - https://github.com/cloudfoundry/cf-release/commit/2ad0317 and https://github.com/cloudfoundry/cf-release/commit/58cdd6d. Are these changes needed here?

(Found by doing `git diff -U5 v256..v257 templates` in cf-release)

**Comment by paroxp** [comment]
I don't think it's a requirement.

> Client authors can now use the Routing API to retrieve a particular router group for a specified name [details](https://www.pivotaltracker.com/n/projects/1358110/stories/142808089)
Source: [routing-release@0.150.0](https://github.com/cloudfoundry-incubator/routing-release/releases/tag/0.150.0)

We may want to throw that into the backlog, to perhaps improve filtering of some sort or create groups for security control?

We should throw a team/tech talk :)

**Comment by dcarley** [comment]
> We should throw a team/tech talk :)

From Slack: this was addressed in e75ae4f because we want to track cf-release changes as closely as possible.

**Pull request merged**